modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+
+Each provider supplies its own master password. When you use a provider to encrypt a new credential, you can only use that provider to recover the encrypted credential.
+
+Once a master password is created, Studio caches it until it is closed, or you manually clear the password from Studio's cache.
+
+== Secure Storage File
+
+This file holds all your user credentials in an encrypted form. By default, the secure storage file is located in `$HOME/.eclipse/org.eclipse.equinox.security/secure_storage`.
+
+Because it is a single file, the same secure storage is shared across all Eclipse-based workspaces and Anypoint Studio instances.
+
+
+== Update the Master Password
+
+You can change or rotate the master password for a particular provider. When changing a master password, all the encrypted data by this provider is re-encrypted with the new master password.
+
+=== Before you Begin
+
+* Ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
+Because the secure storage file is shared across all running instances and the master password is cached in each instance's memory, updating your master password causes other opened instances to still write the credentials using the old master password. Closing all other instances ensures you avoid the <<no-password-provided-exception,No Password Provided Exception>>.
+
+To re-generate your master password:
+
+. Navigate to *Preferences* > *General* > *Security* > *Secure Storage*.
+. Select *Change Password...*:
++
+image::secure-storage-change-password.png[]
+. Studio prompts you to create a password hint for this new password. +
+Select *Yes* to generate a new password hint:
++
+image::new-password-created.png[]
++
+[CAUTION]
+--
+If you don't supply new password hints, then the former ones persist, which reference the former password. You cannot recover this new password using the previous password hints.
+--
+. Studio prompts you that the password has been successfully updated.
+. Select *Apply and Close*.
+
+
+== Customize the Location of the Secure Storage and Password Files
+
+To change the location of the password and secure storage files:
+
+. Locate your `AnypointStudio.ini` file:
+* In Windows:
+.. Go to your Anypoint Studio's installation directory.
+* In macOS:
+.. Go to your `/Applications` folder, right-click the `AnypointStudio` package and select *Show Package Contents*.
+.. Go to `Contents` > `Eclipse`, and open the `AnypointStudio.ini` file.
+* In Linux:
+.. Go to `/usr/lib/anypointstudio/anypointstudio.ini`.
+. Add the following options together to bypass the password provider mechanism:
++
+[source]
+--
+-eclipse.keyring <file_path> //<1>
+-eclipse.password <file_path> //<2>
+--
+<1> Configures the location of the Secure Storage file.
+<2> Configures the location of the password file. Studio uses this file's content as the master password.
+
```

Comment:
Was the omission of the another alternative (via arguments in the command line) on purpose?